### ISSUE TITLE
Feature request: dedupe emitted types referenced from multiple files

### ISSUE THREAD

**Issue by mridang** [bcherny/json-schema-to-typescript#333]
I have two schema files in a directory.

```
{
  "type" : "object",
  "id" : "WebsiteOrder.json",
  "title" : "WebsiteOrder",
  "properties" : {
    "external_order_ref" : {
      "type" : "string"
    },
    "info" : {
      "type" : "object",
      "$ref" : "./res/OrderCustomer.schema#",
      "required" : true
    }
  }
}
```

and 

```
{
  "type" : "object",
  "id" : "OrderCustomer",
  "title" : "OrderCustomer",
  "properties" : {
    "first_name" : {
      "type" : "string"
    },
    "last_name" : {
      "type" : "string"
    },
    "email" : {
      "type" : "string"
    },
    "phone" : {
      "type" : "string"
    },
    "post_code" : {
      "type" : "string"
    },
    "country" : {
      "type" : "string"
    },
    "newsletter" : {
      "type" : "string"
    },
    "order_number" : {
      "type" : "string",
      "required" : true
    },
    "type" : {
      "type" : "string"
    }
  }
}
```

When I run the command to generate TypeScript types for those schema files, I end up with duplicates in my resultant file. I end up with two files `WebsiteOrder.d.ts`:

```
export interface WebsiteOrder {
  external_order_ref?: string;
  info?: OrderCustomer;
  [k: string]: unknown;
}
export interface OrderCustomer {
  first_name?: string;
  last_name?: string;
  email?: string;
  phone?: string;
  post_code?: string;
  country?: string;
  newsletter?: string;
  order_number?: string;
  type?: string;
  [k: string]: unknown;
}
```

and a `OrderCustomer.d.ts`

```
export interface OrderCustomer {
  first_name?: string;
  last_name?: string;
  email?: string;
  phone?: string;
  post_code?: string;
  country?: string;
  newsletter?: string;
  order_number?: string;
  type?: string;
  [k: string]: unknown;
}

```

Is there support (or documentation) or how to unify the resultant types?

**Comment by NathanJAdams** [comment]
This is affecting me too, I have a complex schema with multiple types that are re-used and would love to be able to re-use the typescript types instead of having multiple copies of the referenced types

**Comment by bcherny** [comment]
Proposals for how to do this welcome! It will require a bit of thinking around how to deal with types referenced by multiple files. As a start, what would you expect the output to be in this case?

**Comment by bcherny** [comment]
Merged into #258
